### ISSUE TITLE
Exploding gradients/weights/logits in a fine-tuning notebook

### ISSUE THREAD

**Issue by 81549361** [bigscience-workshop/petals#283]
Hello, after I successfully fine-tuned, I run to the last step and when generating text, the following error occurs. 
May I ask what is the reason?

![image](https://user-images.githubusercontent.com/7599454/224216132-416b7044-dee1-4401-a7c7-c2f76ea37660.png)
![image](https://user-images.githubusercontent.com/7599454/224216173-81b75b67-79d0-4b6f-a654-2cf500ea07d5.png)

**Comment by borzunov** [comment]
Hi @81549361, sorry for the long response.

One possible reasons is that model weights/gradients have exploded during training. You would see this on the loss function plot in wandb.

We'll double-check training hyperparams and fix them to avoid this behavior (cc @justheuristic).

**Comment by mryab** [comment]
Hi @81549361, we've just pushed a commit that should fix the explosion in https://github.com/bigscience-workshop/petals/pull/343. Can you try running the updated SST-2 prompt tuning notebook from the main branch?
